engine/src/main/java/org/hibernate/validator/spi/property/JavaBeanPropertySelector.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.spi.property;
+
+import org.hibernate.validator.internal.util.ReflectionHelper;
+
+import java.lang.reflect.Method;
+
+/**
+ * Default property accessor which follows java bean naming convention.
+ */
+public class JavaBeanPropertySelector implements PropertyAccessorSelector {
+
+	@Override
+	public MethodType methodType( Method method ) {
+		return ReflectionHelper.isGetterMethod( method ) ?  MethodType.PROPERTY_READ_ACCESSOR : MethodType.OTHER;
```

Comment:
Btw, it looks like we don't need this method at all. Let's simply have a `isGetterMethod()` declared and remove the `methodType()` method altogether (and we don't need the default method in the interface then).